flutter_app/test/widget/return_to_launch_widget_test.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:imacs/modules/mavlink_communication.dart';
+import 'package:imacs/modules/return_to_launch.dart';
+import 'package:imacs/widgets/return_to_launch_widget.dart';
+
+void main() {
+  group("Return to Launch Widget", () {
+    testWidgets("Return to Launch Button Generate",
+        (WidgetTester tester) async {
+      final mavlinkCommunication = MavlinkCommunication(
+          MavlinkCommunicationType.tcp, '127.0.0.1', 14550);
+      final ReturnToLaunch command = ReturnToLaunch(comm: mavlinkCommunication);
+
+      // Tests to see if the button renders
+      await tester.pumpWidget(MaterialApp(
+          home: Scaffold(
+              body: ReturnToLaunchButton(
+                  returnToLaunchCommand: command,
+                  systemID: 1,
+                  componentID: 0))));
+
+      // Waits for all frames and animations to settle
+      await tester.pumpAndSettle();
+
+      expect(find.byType(ElevatedButton), findsOneWidget);
+      expect(find.text("Return To Launch"), findsOneWidget);
+    });
+
+    testWidgets("Button sends MavLink command", (WidgetTester tester) async {
+      final mavlinkCommunication = MavlinkCommunication(
+          MavlinkCommunicationType.tcp, '127.0.0.1', 14550);
+      final ReturnToLaunch command = ReturnToLaunch(comm: mavlinkCommunication);
+
+      // Tests to see if the button renders
+      await tester.pumpWidget(MaterialApp(
+          home: Scaffold(
+              body: ReturnToLaunchButton(
+                  returnToLaunchCommand: command,
+                  systemID: 1,
+                  componentID: 0))));
+
+      await tester.pumpAndSettle();
+      await tester.tap(find.byType(ReturnToLaunchButton));
+      await tester.pumpAndSettle();
+
+      expect(find.text("Return to Launch Command Sent"), findsOneWidget);
```

Comment:
There isn't really a perfect way to check this - UI is weird :'). I'm guess you copied this over from the test for set drone more, this works over there since when the drone mode is changed, the text displayed also changes (in the widget not the log). Low key this test isn't that important since the actual widget part is subject to change. If you still want to work on this try adding a way for you module to communicate with the rest of the program that the command has been sent.